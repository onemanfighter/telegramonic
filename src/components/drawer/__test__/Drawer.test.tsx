import { render, screen } from '@testing-library/react';
import DrawerComponent from '../Drawer';

describe('Drawer', () => {
  const onModalClose = jest.fn();

  beforeEach(() => {
    onModalClose.mockReset();
  });

  it('should render correctly if isOpen is true', () => {
    const { container } = render(
      <DrawerComponent
        title="title"
        isOpen={true}
        onModalClose={onModalClose}
        children={<div>Children</div>}
        footer={<div>Footer</div>}
      />,
    );

    expect(screen.findByText('title')).toBeDefined();
    expect(screen.findByText('Children')).toBeDefined();
    expect(screen.findByText('Footer')).toBeDefined();
  });

  it('should not render if isOpen is false', () => {
    const { container } = render(
      <DrawerComponent
        title="title"
        isOpen={false}
        onModalClose={onModalClose}
        children={<div>Children</div>}
        footer={<div>Footer</div>}
      />,
    );

    expect(screen.queryByText('title')).toBeNull();
    expect(screen.queryByText('Children')).toBeNull();
    expect(screen.queryByText('Footer')).toBeNull();
    expect(container).toMatchSnapshot();
  });
});
