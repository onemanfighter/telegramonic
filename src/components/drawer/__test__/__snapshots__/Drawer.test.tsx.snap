// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer should not render if isOpen is false 1`] = `<div />`;
